tests-behat/bootstrap/Context.php

```diff
@@ -419,6 +430,139 @@ public function iClickIconUsingCss($arg1)
+     */
+    public function scopeBuilderRule($name, $operator, $value)
+    {
+        $rule = $this->assertScopeBuilderRuleExist($name);
+        $this->assertSelectedValue($rule, $operator, '.vqb-rule-operator select');
+        $this->assertInputValue($rule, $value);
+    }
+
+    /**
+     * hasOne reference or enum type rule for ScopeBuilder.
+     *
+     * @Then /^reference rule "([^"]*)" operator is "([^"]*)" and value is "([^"]*)"$/
+     */
+    public function scopeBuilderReferenceRule($name, $operator, $value)
+    {
+        $rule = $this->assertScopeBuilderRuleExist($name);
+        $this->assertSelectedValue($rule, $operator, '.vqb-rule-operator select');
+        $this->assertSelectedValue($rule, $value, '.vqb-rule-input select');
+    }
+
+    /**
+     * Date, Time or Datetime rule for ScopeBuilder.
+     *
+     * @Then /^date rule "([^"]*)" operator is "([^"]*)" and value is "([^"]*)"$/
+     */
+    public function scopeBuilderDateRule($name, $operator, $value)
+    {
+        $rule = $this->assertScopeBuilderRuleExist($name);
+        $this->assertSelectedValue($rule, $operator, '.vqb-rule-operator select');
+        $this->assertInputValue($rule, $value, 'input.form-control');
+    }
+
+    /**
+     * Boolean type rule for ScopeBuilder.
+     *
+     * @Then /^bool rule "([^"]*)" has value "([^"]*)"$/
+     */
+    public function scopeBuilderBoolRule($name, $value)
+    {
+        $this->assertScopeBuilderRuleExist($name);
+        $idx = ($value === 'Yes') ? 0 : 1;
+        $isChecked = $this->getSession()->evaluateScript('return $(\'[data-name="' . $name . '"]\').find(\'input\')[' . $idx . '].checked');
+        if (!$isChecked) {
+            throw new \Exception('Radio value selected is not: ' . $value);
+        }
+    }
+
+    /**
+     * @Then /^I check if word from data scope match$/
+     */
+    public function iCheckIfWordMatch()
+    {
+        $pageWordContainer = $this->getSession()->getPage()->find('css', '.atk-expected-word-result .content');
+        if (!$pageWordContainer) {
+            throw new \Exception('Unable to find expected word in page.');
+        }
+
+        $expectedWord = $pageWordContainer->getText();
+
+        $wordResponseContainer = $this->getSession()->getPage()->find('css', '.atk-callback-response .content');
+        if (!$wordResponseContainer) {
+            throw new \Exception('Unable to find response word in page.');
+        }
+
+        $responseWord = $wordResponseContainer->getText();
+
+        if (preg_replace('/\s*/m', '', $expectedWord) !== preg_replace('/\s*/m', '', $responseWord)) {
+            throw new \Exception('Data word does not match: ' . $responseWord . ' expected: ' . $expectedWord);
+        }
+    }
+
+    /**
+     * @Then /^I check if input "([^"]*)" match$/
+     */
+    public function iCheckIfInputMatch($name)
+    {
+        $expected = $this->getSession()->getPage()->find('css', '.atk-expected-input-result .content')->getText();
```

Comment:
I do not see any advantage of this, we should just `I check if input XXX match YYY`

Reply:
Because it will test if ScopeBuilder is properly building Vue Query Builder (VQB) query. VQB json query object is set by ScopeBuilder using Model condition. It should match value predefine by condition set in test file.

Reply:
If you pass the target as selector (like in the next comment), I am ok with it, but Context class should not grow without significant reason for new API.